packages/cli/src/commands/hydrogen/upgrade.ts

```diff
@@ -0,0 +1,1119 @@
+      const pr = item.pr ? `[#${item.id}](${item.pr})\n` : '';
+      const multiStep = steps.length > 1;
+      const title = multiStep
+        ? `#### Step: ${stepIndex + 1}. ${step.title} ${pr}\n`
+        : `#### ${step.title.trim()}\n`;
+      const info = step.info ? `> ${step.info}\n` : '';
+      const code = step.code ? `${Buffer.from(step.code, 'base64')}\n` : '';
+      const docs = item.docs ? `[docs](${item.docs})\n` : '';
+      return `${title}${info}${docs}${pr}${code}`;
+    })
+    .join('\n');
+  return `${heading}\n${body}`;
+}
+
+/**
+ * Generates a markdown file with upgrade instructions
+ */
+async function generateUpgradeInstructionsFile({
+  appPath,
+  cumulativeRelease,
+  currentVersion,
+  dryRun,
+  selectedRelease,
+}: {
+  appPath: string;
+  cumulativeRelease: CumulativeRelease;
+  currentVersion: string;
+  dryRun: boolean;
+  selectedRelease: Release;
+}) {
+  let filename = '';
+
+  const {featuresMd, breakingChangesMd} = cumulativeRelease.features
+    .filter((feature) => feature.steps)
+    .reduce(
+      (acc, feature) => {
+        if (feature.breaking) {
+          acc.breakingChangesMd.push(generateStepMd(feature));
+        } else {
+          acc.featuresMd.push(generateStepMd(feature));
+        }
+        return acc;
+      },
+      {featuresMd: [], breakingChangesMd: []} as {
+        featuresMd: string[];
+        breakingChangesMd: string[];
+      },
+    );
+
+  const fixesMd = cumulativeRelease.fixes
+    .filter((fixes) => fixes.steps)
+    .map(generateStepMd);
+
+  if (!featuresMd.length && !fixesMd.length) {
+    renderInfo({
+      headline: `No upgrade instructions generated`,
+      body: `There are no additional upgrade instructions for this version.`,
+    });
+    return;
+  }
+
+  const absoluteFrom = getAbsoluteVersion(currentVersion);
+  const absoluteTo = getAbsoluteVersion(selectedRelease.version);
+  filename = `${
+    dryRun ? 'preview-' : ''
+  }upgrade-${absoluteFrom}-to-${absoluteTo}.md`;
+
+  const dotShopifyFolderPath = path.join(appPath, '.shopify/');
+
+  const h1 = `# Hydrogen upgrade guide: ${absoluteFrom} to ${absoluteTo}`;
+
+  let md = `${h1}\n\n----\n`;
+
+  if (breakingChangesMd.length) {
+    md += `\n## Breaking changes\n\n${breakingChangesMd.join('\n')}\n----\n`;
+  }
+
+  if (featuresMd.length) {
+    md += `\n## Features\n\n${featuresMd.join('\n')}\n----\n`;
+  }
+
+  if (fixesMd.length) {
+    md += `\n${featuresMd.length ? '----\n\n' : ''}## Fixes\n\n${fixesMd.join(
+      '\n',
+    )}`;
+  }
+
+  const filePath = path.join(dotShopifyFolderPath, filename);
+
+  try {
+    await isDirectory(dotShopifyFolderPath);
+  } catch (error) {
+    await mkdir(dotShopifyFolderPath);
+  }
+
+  if (!(await fileExists(filePath))) {
+    await touchFile(filePath);
+  } else {
+    const overwriteMdFile = await renderConfirmationPrompt({
+      message: `A previous upgrade instrunctions file already exists for this version.\nDo you want to overwrite it?`,
```

Comment:
Noticed this in the CLI and went searching for it, there's a few misspellings of `instructions` in this file!

```suggestion
      message: `A previous upgrade instructions file already exists for this version.\nDo you want to overwrite it?`,
```